lib/rubygoal/coach.rb

```diff
@@ -16,30 +16,79 @@ def formation(match)
       raise NotImplementedError
     end
 
-    def players_errors
-      errors = {}
+    def errors
+      @errors = []
 
-      config = Rubygoal.configuration
+      check_unique_captain
+      check_players_count(:average)
+      check_players_count(:fast)
 
+      @errors
+    end
```

Comment:
You're so right! :dart: (your initial guess it's correct: I don't want to cache the errors result at all, so using a class attribute is strange)